packages/target-tools/src/transforms/requestTransform.js

```diff
@@ -0,0 +1,438 @@
+            target: {
+              ...targetParams,
+              profile: {
+                ...targetProfileParams
+              }
+            }
+          }
+        }
+      };
+
+      const lastEvent = events[events.length - 1];
+
+      if (
+        events.length > 0 &&
+        isRoughlyTheSameObject(lastEvent.data, event.data)
+      ) {
+        lastEvent.query.personalization.decisionScopes.push(
+          ...event.query.personalization.decisionScopes
+        );
+      } else {
+        events.push(event);
+      }
+    });
+
+  return {
+    ...edgeRequest,
+    events
+  };
+}
+
+/**
+ *
+ * @param { import("@adobe/aep-edge-tools/aep-edge-api-client/models/EdgeRequest").EdgeRequest } edgeRequest
+ * @returns { import("@adobe/aep-edge-tools/aep-edge-api-client/models/EdgeRequest").EdgeRequest }
+ */
+function addIdentityQuery(edgeRequest) {
+  const { query = {} } = edgeRequest;
+  const { identity = {} } = query;
+  const { fetch = [] } = identity;
+
+  return {
+    ...edgeRequest,
+    query: {
+      ...query,
+      identity: {
+        ...identity,
+        fetch: Array.from(new Set([...fetch, "ECID", "TNTID"]))
+      }
+    }
+  };
+}
+
+/**
+ *
+ * @param { import("@adobe/aep-edge-tools/aep-edge-api-client/apis/InteractApi").InteractPostRequest } interactRequest
+ * @param { import("../../delivery-api-client/models/DeliveryRequest").DeliveryRequest } deliveryRequest
+ * @returns { import("@adobe/aep-edge-tools/aep-edge-api-client/apis/InteractApi").InteractPostRequest }
+ */
+function createEdgeRequest(interactRequest, { deliveryRequest }) {
+  const edgeRequestPipeline = createPipeline([
+    translateIdentities,
+    translateTargetContext,
+    translateExecuteRequest,
+    addIdentityQuery
+  ]);
+
+  return {
+    ...interactRequest,
+    edgeRequest: edgeRequestPipeline.execute(
+      {
+        xdm: {}
+      },
+      deliveryRequest
+    )
+  };
+}
+
+/**
+ *
+ * @param { import("@adobe/aep-edge-tools/aep-edge-api-client/apis/InteractApi").InteractPostRequest } interactRequest
+ * @param { String } imsOrgId
+ * @param { String } sessionId
+ * @param { import("../../delivery-api-client/models/DeliveryRequest").DeliveryRequest } deliveryRequest
+ * @returns { import("@adobe/aep-edge-tools/aep-edge-api-client/apis/InteractApi").InteractPostRequest }
+ */
+function addMetaEntries(
+  interactRequest,
+  { imsOrgId, sessionId, deliveryRequest, konductorIdentity }
+) {
+  const { context = {} } = deliveryRequest;
+  const { browser = {}, address = {} } = context;
+
+  const { host } = browser;
+  const { url } = address;
+
+  const { edgeRequest } = interactRequest;
+  const { meta = {} } = edgeRequest;
+  const { state = {} } = meta;
+  const { entries = [] } = state;
+
+  const { host: domain = "" } = parseURI(host || url || "") || {};
+
+  const additionalState = {
+    domain,
+    cookiesEnabled: true,
+    entries: [...entries]
+  };
+
+  if (isNotBlank(imsOrgId) && isNotBlank(sessionId)) {
```

Comment:
extract method

Reply:
what are you trying to achieve by extracting a method here?  in general, i prefer to extract methods for readability or code reuse.  this logic check is only used in this one place.  is it hard to read?

Reply:
Just trying to improve readability and reduce method length by abstracting away chunks of code that can be isolated in separate methods. By replacing these 6 lines of code with a simple call to `setSessionCookie` the method becomes more readable.